rust/agama-cli/src/profile.rs

```diff
@@ -114,48 +115,59 @@ async fn import(url_string: String, dir: Option<PathBuf>) -> anyhow::Result<()>
     });
+
     let url = Url::parse(&url_string)?;
     let tmpdir = TempDir::new()?; // TODO: create it only if dir is not passed
+    let work_dir = dir.unwrap_or_else(|| tmpdir.into_path());
+    let profile_path = work_dir.join("profile.json");
+
+    // Specific AutoYaST handling
     let path = url.path();
-    let output_file = if path.ends_with(".sh") {
-        "profile.sh"
-    } else if path.ends_with(".jsonnet") {
-        "profile.jsonnet"
-    } else {
-        "profile.json"
-    };
-    let output_dir = dir.unwrap_or_else(|| tmpdir.into_path());
-    let mut output_path = output_dir.join(output_file);
-    let output_fd = File::create(output_path.clone())?;
     if path.ends_with(".xml") || path.ends_with(".erb") || path.ends_with('/') {
-        // autoyast specific download and convert to json
-        AutoyastProfile::new(&url)?.read_into(output_fd)?;
+        // AutoYaST specific download and convert to JSON
+        AutoyastProfileImporter::read(&url)?.write_file(&profile_path)?;
     } else {
-        // just download profile
-        Transfer::get(&url_string, output_fd)?;
+        pre_process_profile(&url_string, &profile_path)?;
     }
 
-    // exec shell scripts
-    if output_file.ends_with(".sh") {
-        let err = Command::new("bash")
-            .args([output_path.to_str().context("Wrong path to shell script")?])
-            .exec();
-        eprintln!("Exec failed: {}", err);
-    }
+    validate(&profile_path)?;
+    store_settings(&profile_path).await?;
 
-    // evaluate jsonnet profiles
-    if output_file.ends_with(".jsonnet") {
-        let fd = File::create(output_dir.join("profile.json"))?;
-        let evaluator = ProfileEvaluator {};
-        evaluator
-            .evaluate(&output_path, fd)
-            .context("Could not evaluate the profile".to_string())?;
-        output_path = output_dir.join("profile.json");
-    }
+    Ok(())
+}
 
-    validate(&output_path)?;
-    store_settings(&output_path).await?;
+// Preprocess the profile.
+//
+// The profile can be a JSON or a Jsonnet file or a script.
+//
+// * If it is a JSON file, no preprocessing is needed.
+// * If it is a Jsonnet file, it is converted to JSON.
+// * If it is a script, it is executed.
+fn pre_process_profile<P: AsRef<Path>>(url_string: &str, path: P) -> anyhow::Result<()> {
+    let work_dir = path.as_ref().parent().unwrap();
+    let tmp_profile_path = work_dir.join("profile.temp");
+    let tmp_file = File::create(&tmp_profile_path)?;
+    Transfer::get(url_string, tmp_file)?;
 
+    match FileFormat::from_file(&tmp_profile_path)? {
+        FileFormat::Jsonnet => {
+            let file = File::create(path)?;
+            let evaluator = ProfileEvaluator {};
+            evaluator
+                .evaluate(&tmp_profile_path, file)
+                .context("Could not evaluate the profile".to_string())?;
+        }
+        FileFormat::Script => {
+            let err = Command::new("bash").args([&tmp_profile_path]).exec();
```

Comment:
As Josef said, we are checking for *any* script, even Python, but then run it with `bash`.
1. Either check for `#!/bin/bash`,
2.  or make it executable and `Command::new(tmp_profile_path)`

Reply:
Yes, that's why I pointed to how we run those scripts after reading @jreidinger's comment . For the scripting support, I followed the 2nd approach (make it executable).

However, when handling the agama.auto we have never considered anything different from bash.

So we can keep the current check and make it executable; or just check for bash.

Reply:
Just to clarify, this line of code [was already there](https://github.com/agama-project/agama/blob/6284cbfebeb4be8b7808120ef8a7d72d91b91569/rust/agama-cli/src/profile.rs#L140). It was just moved in this PR.